test/framework/machine_helpers.go

```diff
@@ -21,32 +21,106 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"github.com/pkg/errors"
 
+	"github.com/pkg/errors"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 )
 
+type GetMachinesByClusterInput struct {
+	Lister      Lister
+	ClusterName string
+	Namespace   string
+}
+
+type GetControlPlaneMachinesByClusterInput struct {
+	GetMachinesByClusterInput
+}
+
+// GetControlPlaneMachinesByCluster returns the control plane Machine objects for a cluster.
+// Important! this method relies on labels that are created by the CAPI controllers during the first reconciliation, so
+// it is necessary to ensure this is already happened before calling it.
+func GetControlPlaneMachinesByCluster(ctx context.Context, input GetControlPlaneMachinesByClusterInput) []clusterv1.Machine {
+	options := append(byClusterOptions(input.ClusterName, input.Namespace), controlPlaneMachineOptions()...)
+
+	machineList := &clusterv1.MachineList{}
+	Expect(input.Lister.List(ctx, machineList, options...)).To(Succeed(), "Failed to list MachineList object for Cluster %s/%s", input.Namespace, input.ClusterName)
+
+	return machineList.Items
+}
+
+type GetMDMachinesByClusterInput struct {
```

Comment:
Can we use MachineDeployment instead of MD (we are not using abbreviations anywhere)
This should probably be `GetMachineByMachineDeployments` (it is implicit a machine deployment belongs to a Cluster)